test/lib/tasks/bundlers/generateComponentPreload.js

```diff
@@ -101,3 +109,490 @@ test.serial("generateComponentPreload", async (t) => {
+					".properties",
+					".json",
+				],
+				name: "my/app2/Component-preload.js",
+				sections: [
+					{
+						filters: [
+							"my/app2/",
+							"!my/app2/test/",
+							"!my/app2/*.html",
+							"!my/app1/thirdparty1/",
+							"!my/app2/thirdparty2/",
+							"+my/app2/thirdparty2/NotExcluded.js"
+						],
+						mode: "preload",
+						renderer: false,
+						resolve: false,
+						resolveConditional: false,
+					}
+				]
+			},
+			bundleOptions: {
+				optimize: true,
+				ignoreMissingModules: true
+			}
+		},
+		resources
+	}]);
+
+	t.is(comboByGlob.callCount, 1,
+		"combo.byGlob should have been called once");
+	t.deepEqual(comboByGlob.getCall(0).args, ["/resources/**/*.{js,json,xml,html,properties,library}"],
+		"combo.byGlob should have been called with expected pattern");
+
+	t.is(ReaderCollectionPrioritizedStub.callCount, 1,
+		"ReaderCollectionPrioritized should have been called once");
+	t.true(ReaderCollectionPrioritizedStub.calledWithNew(),
+		"ReaderCollectionPrioritized should have been called with 'new'");
+
+	const bundleResources1 = await moduleBundlerStub.getCall(0).returnValue;
+	const bundleResources2 = await moduleBundlerStub.getCall(1).returnValue;
+	t.is(workspace.write.callCount, 2,
+		"workspace.write should have been called twice");
+	t.deepEqual(workspace.write.getCall(0).args, [bundleResources1[0]],
+		"workspace.write should have been called with expected args");
+	t.is(workspace.write.getCall(0).args[0], bundleResources1[0],
+		"workspace.write should have been called with exact resource returned by moduleBundler");
+	t.deepEqual(workspace.write.getCall(1).args, [bundleResources2[0]],
+		"workspace.write should have been called with expected args");
+	t.is(workspace.write.getCall(1).args[0], bundleResources2[0],
+		"workspace.write should have been called with exact resource returned by moduleBundler");
+});
+
+test.serial("generateComponentPreload - one namespace - invalid exclude", async (t) => {
+	const {
+		generateComponentPreload,
+		workspace, dependencies, comboByGlob,
+		log
+	} = t.context;
+
+	const resources = [
+		{"fake": "resource"}
+	];
+	comboByGlob.resolves(resources);
+
+	await generateComponentPreload({
+		workspace,
+		dependencies,
+		options: {
+			projectName: "Test Application",
+			namespaces: ["my/app"],
+			excludes: [
+				"!**/" // re-include outside of namespace is not allowed
+			]
+		}
+	});
+
+	t.is(log.warn.callCount, 1, "log.warn should be called once");
+	t.deepEqual(log.warn.getCall(0).args, [
+		"Configured preload exclude contains invalid re-include: !**/. " +
+		"Re-includes must start with a component namespace (my/app)"
+	]);
+
+	t.is(log.verbose.callCount, 1, "log.verbose should be called once");
+	t.deepEqual(log.verbose.getCall(0).args, [
+		"Generating my/app/Component-preload.js..."
+	]);
+
+	t.is(log.error.callCount, 0, "log.error should not be called");
+});
+
+test.serial("generateComponentPreload - nested namespaces - excludes", async (t) => {
```

Comment:
Perhaps write 'minus' instead of the minus sign? I find it quicker to understand

Reply:
Since this is an internal test file I would leave it as is. The use of the minus sign as a separator is consistent with the other tests in the file.

Reply:
OK, I'll leave the internal test files alone in the future, unless you ask me to look at them specifically.